bliss/simulator/decoder.py

```diff
@@ -14,55 +14,49 @@
 class ImageDecoder(nn.Module):
     def __init__(
         self,
-        n_bands: int,
         pixel_scale: float,
-        psf_params_file: str,
         psf_slen: int,
-        sdss_bands: Tuple[int, ...],
+        sdss_fields: DictConfig,
     ) -> None:
         super().__init__()
-        assert n_bands == 1, "Only 1 band is supported"
 
-        self.n_bands = n_bands
+        self.n_bands = len(sdss_fields["bands"])
         self.pixel_scale = pixel_scale
+        self.psf_slen = psf_slen
+        self.psf_galsim = {}  # Dictionary indexed by (run, camcol, field) tuple
 
-        self.params = None  # psf params from fits file
+        sdss_dir = sdss_fields["dir"]
```

Comment:
Here, we retrieve the relevant information for computing the relative flux values for sampled light sources from arbitrary SDSS catalog rcf's.